eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/log/EventLog.scala

```diff
@@ -596,7 +596,8 @@ abstract class EventLog[A <: EventLogState](id: String) extends Actor with Event
       case (acc, e) =>
         snr += 1
 
-        val e2 = e.prepare(id, snr, e.systemTimestamp)
+        val timestamp = if (e.systemTimestamp == 0L) currentSystemTime else e.systemTimestamp
```

Comment:
Pass the current system time via a `systemTimestamp` parameter as done in `prepareEvents`. All events of an atomically written batch should have the same system time.